shell/platform/android/io/flutter/embedding/android/FlutterFragment.java

```diff
@@ -1117,8 +1189,25 @@ public boolean shouldRestoreAndSaveState() {
     return true;
   }
 
+  /**
+   * {@inheritDoc}
+   *
+   * <p>Be wary of overriding this method when using {@code
+   * enableAutomaticOnBackPressedHandling(true)}. The general recommendation is to avoid overriding
+   * it in that case, but if you do, always {@code return super.popSystemNavigator()} rather than
+   * {@code return false}. Otherwise the navigation behavior will break.
+   */
   @Override
   public boolean popSystemNavigator() {
+    if (getArguments().getBoolean(ARG_ENABLE_AUTOMATIC_ON_BACK_PRESSED_HANDLING, false)) {
+      FragmentActivity activity = getActivity();
+      if (activity != null) {
+        onBackPressedCallback.setEnabled(false);
```

Comment:
Add comment and/or link to bugs so the next reviewer doesn't accidentally "optimize" it away.

Reply:
Done